From: Suresh Jayaraman <sjayaraman@suse.com>
Subject: SUNRPC: increase maximum slots to use
References: bnc#775685
Patch-mainline: No [upstream approach breaks kABI]

Note: Upstream had adopted dynamic slot allocation for TCP connections. The
upstream patchset will break kABI.

The current RPC max slot table size of 128 might be too limiting on GigE
links. This patch increases the rpc maximum slots to use to 1024 from 128.
This should allow for better performance on 10GigE links.

Also note that this patch needs no forward-porting/maintenance as upstream has
similar effects with dynamic rpc slot patches.

Signed-off-by: Suresh Jayaraman <sjayaraman@suse.com>
---

Index: linux-3.0-SLE11-SP2/include/linux/sunrpc/xprt.h
===================================================================
--- linux-3.0-SLE11-SP2.orig/include/linux/sunrpc/xprt.h
+++ linux-3.0-SLE11-SP2/include/linux/sunrpc/xprt.h
@@ -21,7 +21,7 @@
 
 #define RPC_MIN_SLOT_TABLE	(2U)
 #define RPC_DEF_SLOT_TABLE	(16U)
-#define RPC_MAX_SLOT_TABLE	(128U)
+#define RPC_MAX_SLOT_TABLE	(1024U)
 
 /*
  * This describes a timeout strategy
Index: linux-3.0-SLE11-SP2/include/linux/nfs_fs_sb.h
===================================================================
--- linux-3.0-SLE11-SP2.orig/include/linux/nfs_fs_sb.h
+++ linux-3.0-SLE11-SP2/include/linux/nfs_fs_sb.h
@@ -182,7 +182,7 @@ struct nfs_server {
 
 
 /* maximum number of slots to use */
-#define NFS4_MAX_SLOT_TABLE RPC_MAX_SLOT_TABLE
+#define NFS4_MAX_SLOT_TABLE (128U)
 
 #if defined(CONFIG_NFS_V4)
 
